src/client/features/interactions/index.js

```diff
@@ -119,7 +156,30 @@ class Interactions extends React.Component {
     const id=this.findId(nodeMetadata,query);
     return {id,network};
   }
+  filterUpdate(type) {
+    const state=this.state;
+    const catagories = state.catagories;
+    const buttons=state.buttons;
+    const cy= state.cy;
+    const edges=catagories.get(type).edges;
+    const nodes=catagories.get(type).nodes;
+    
+    hideTooltips(cy);
+    const hovered = cy.filter(ele=>ele.style('background-color')==='blue'||ele.style('line-color')==='orange');
```

Comment:
This is wrong.  You're conflating style with state.  Use flags or other real values for meaningful state.  The style should just be mapped from the state using the stylesheet.

Reply:
Consider, for example, that the colours may be changed in future and this code would break.